src/components/ProductBacklogListComponent.js

```diff
@@ -44,71 +48,121 @@ const ProductBacklogListComponent = () => {
+      console.log(project);
+      await updateProject();
+    }
+  }, [state.productBacklog]);
+
+  const addNewStory = async (story) => {
     setState({ productBacklog: [...state.productBacklog, story] });
+    //await updateProject()
+  };
+
+  const updateProject = async () => {
+    //Update the db
+    try {
+      let result = await db.updateProject({
+        projectName: project.projectName,
+        productBacklog: state.productBacklog,
+      });
+
+      //Check if the update worked
+      if (!result.success) throw { message: result.message };
+
+      await refreshProjects(); //Refresh the projects stored as they were modified
+      displayPopup('Successfully saved new user story');
+    } catch (err) {
+      displayPopup(`Error saving user story ${err.message}`);
+
+      console.log(`Project update failed: ${err}`);
+    }
   };
 
-  const onDeleteItem = (item) => {
-    setState({
+  const onDeleteItem = async (item) => {
+    await setState({
       productBacklog: state.productBacklog.filter((story) => story !== item),
     });
+
+    await updateProject();
   };
 
   const onStoryClick = (item) => {
     //Enable editing the story
-    setState({ editMode: true });
+    setState({ addMode: true, isEditing:true, editingIndex:state.productBacklog.findIndex((story) => story===item) });
+    console.log("Now editing "+state.productBacklog.findIndex((story) => story===item));
+    setNewStory(item);
   };
 
   const renderList = () => {
-    return state.productBacklog.map((item) => (
-      <ListItem button onClick={() => onStoryClick(item)}>
-        <ListItemText
-          primary={item.priority}
-          style={{ maxWidth: '10px', marginRight: '50px' }}
-        />
-        <ListItemText primary={item.task} style={{ flex: 12 }} />
-        <ListItemSecondaryAction>
-          <IconButton
-            edge="end"
-            aria-label="delete"
-            onClick={() => onDeleteItem(item)}
-          >
-            <DeleteIcon />
-          </IconButton>
-        </ListItemSecondaryAction>
-      </ListItem>
-    ));
+    if(state.productBacklog){
+      let tempArray= state.productBacklog.sort((p1,p2) => p1.priority-p2.priority);
+      return tempArray.map((item) => {
+      if (item)
+        //Check if there are any products
+        return (
+          <ListItem button onClick={() => onStoryClick(item)}>
+            <ListItemText
+              primary={item.priority}
+              style={{ maxWidth: '10px', marginRight: '50px' }}
+            />
+            <ListItemText primary={item.task} style={{ flex: 12 }} />
+            <ListItemSecondaryAction>
+              <IconButton
+                edge="end"
+                aria-label="delete"
+                onClick={() => onDeleteItem(item)}
+              >
+                <DeleteIcon style={{fill: "white"}}/>
+              </IconButton>
+            </ListItemSecondaryAction>
+          </ListItem>
+        );
+    });
+  } else 
+    return;
   };
```

Comment:
you don't really need an else return if it's the end of the function, you can just have the if block unless you wanted some sort of logging or something in an else

Reply:
Eh fair enough fair enough, at the time I was just putting it in because I wasnt sure if I wanted to do some logging. Cleaned up now